### PR TITLE
Add GameType, FormatType and Spectator to MatchInfo + Fix getting Bra…

### PR DESCRIPTION
**Pull request by bmichotte**
…wlSeasonId in MatchInfo + Fix GetBrawlInfo

### CODE REVIEW

File: TestAppMacOS/main.m [hunk 1]
```diff
@@ -142,19 +142,19 @@ int main(int argc, const char * argv[]) {
...
+        //loadGameType(mirror);
+        //loadFormat(mirror);
+        //loadMatchInfo(mirror);
+        //loadDecks(mirror);
+        //loadSelectedDeck(mirror);
+        //loadArenaInfo(mirror);
+        //loadArenaDraftChoice(mirror);
+        //loadPackCards(mirror);
+        //loadBrawlInfo(mirror);
+        //loadEditedDeck(mirror);
```
Comment (jleclanche) [inline TestAppMacOS/main.m]: Can you drop the unnecessary diff please

Comment (bmichotte) [inline TestAppMacOS/main.m]: nope, comments are wanted in the _test_ app

Comment (jleclanche) [inline TestAppMacOS/main.m]: you're removing a type of comment and adding a different type of comment, it's just diff noise

Comment (bmichotte) [inline TestAppMacOS/main.m]: yep, but it's boring, when you test a method to change comment type. This way, you only have to uncomment a single line
